### PR TITLE
feat: adding changes to AgentBaker for Stateless CNI on Windows

### PR DESCRIPTION
**Pull request by behzad-mir**
**What type of PR is this?**

**What this PR does / why we need it**:
We are adding a new CNI related file for capturing the states and need to 1) exclude it from MS Defender and 2) add it to the cleanup process during restart.
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: staging/cse/windows/kubernetesfunc.ps1 [hunk 1]
```diff
@@ -144,6 +144,7 @@ function Update-DefenderPreferences {
     Add-MpPreference -ExclusionProcess "C:\k\azurecni\bin\azure-vnet.exe"
     Add-MpPreference -ExclusionProcess "C:\k\azurecni\bin\AzureNetworkContainer.exe"
     Add-MpPreference -ExclusionProcess "C:\k\azurecni\bin\CnsWrapperService.exe"
+    Add-MpPreference -ExclusionProcess "C:\k\azurecns\azure-endpoints.json"
```
Comment (AbelHu) [inline staging/cse/windows/kubernetesfunc.ps1:147]: Does C:\k\azurecns exist?

Comment (behzad-mir) [reply staging/cse/windows/kubernetesfunc.ps1:147]: It does exist in our CNIv2 (Swift)
It does not exist in CNIV1.

Comment (behzad-mir) [reply staging/cse/windows/kubernetesfunc.ps1:147]: Is this going to create issues if this script runs on CNIV1?

Comment (AbelHu) [reply staging/cse/windows/kubernetesfunc.ps1]: I think that we need to test it. If it causes issues, we only can add this with CNIv2.

Comment (AbelHu) [reply staging/cse/windows/kubernetesfunc.ps1]: I run `Add-MpPreference -ExclusionProcess "C:\k\azurecns\azure-endpoints.json"` in a WS2019 node and it has no issue when `C:\k\azurecns\` does not exist

**Review by AbelHu — APPROVED** [review]

**Pull request merged**